server/migrations/20210922101959-create-categories.js

```diff
@@ -0,0 +1,49 @@
+'use strict'
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    return queryInterface.createTable('categories', {
+      id: {
+        autoIncrement: true,
+        primaryKey: true,
+        type: Sequelize.INTEGER,
+      },
+      catname: {
+        type: Sequelize.TEXT,
+        allowNull: false,
+      },
+      createdAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
+      },
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
+      },
+      agencyId: {
+        allowNull: true, //to change to false once we implement this across all agencies
+        type: Sequelize.INTEGER,
+        references: {
+          model: 'agencies',
+          key: 'id',
+        },
+        onUpdate: 'CASCADE',
+        onDelete: 'SET NULL', //Cascade?
+      },
+      parentId: {
+        allowNull: true,
+        type: Sequelize.INTEGER,
+        references: {
+          model: 'categories',
+          key: 'id',
+        },
+        onUpdate: 'CASCADE',
+        onDelete: 'SET NULL', //Cascade?
```

Comment:
Have you resolved if this should be set null or cascade?

Reply:
should be `SET NULL`. I don't think it's a good idea if the parent gets deleted, leaving orphans in its wake.